### DISCUSSION TITLE
Mean losses for NER nlp.update( )

### DISCUSSION THREAD

**Discussion by davesuketu215** [explosion/spaCy#8053] [Help: Coding & Implementations] [answered]
I am performing validation of my custom spaCy NER model to check for convergence. Where I am checking for convergence of Loss values on Training and Validation Sets.

I am using the below snippet for training, as per the standard practice suggested for spaCy NER.
```
lossesValue = { }        
for batch in batches:
     nlp.update(batch, sgd=optimizer, drop=dropoutRate, losses=lossesValue)
print(lossesValue)     
```
I noticed that the loss values output out of this is a very huge floating value (in millions) and consequently do not convey information about convergence. Basically I have lakhs of records for training.
So my question is, looking at the values I doubt whether they are already mean loss values over the entire example set. Are they?

Hence, will it be correct to manually divide the ```lossesValue``` by the number of examples? I observe that dividing this way gives me a better visualisation of the losses and the learning curve, and fits well.

**Comment by svlandeg** [comment, marked as answer]
Your findings are indeed correct, cf this commit by Matt: https://github.com/explosion/spaCy/commit/a4164f67cac6388b16707e6c7dcc9100cd8926e7 (the actual implemenation now lives in https://github.com/explosion/spaCy/blob/master/spacy/pipeline/transition_parser.pyx).

In general, I'd very much recommend looking at Accuracy/P-R-F values on the development data rather than raw loss scores, which aren't very meaningful/interpretable by themselves.
